functional_tests/kfp/kfp-pipeline-helper.py

```diff
@@ -0,0 +1,246 @@
+import os
+import random
+from typing import NamedTuple
+
+import kfp
+from kfp import components
+import kfp.dsl as dsl
+from kubernetes.client.models import V1EnvVar
+from wandb_probe import wandb_probe_package
+
+
+def add_wandb_env_variables(op):
+    env = {
+        "WANDB_API_KEY": os.getenv("WANDB_API_KEY"),
+        "WANDB_BASE_URL": os.getenv("WANDB_BASE_URL"),
+        "WANDB_KUBEFLOW_URL": os.getenv("WANDB_KUBEFLOW_URL"),
+        "WANDB_PROJECT": "wandb_kfp_integration_test",
+    }
+
+    for name, value in env.items():
+        op = op.add_env_variable(V1EnvVar(name, value))
+    return op
+
+
+def preprocess_data(
+    X_train_path: components.OutputPath("np_array"),  # noqa: F821,N803
+    X_test_path: components.OutputPath("np_array"),  # noqa: F821,N803
+    y_train_path: components.OutputPath("np_array"),  # noqa: F821
+    y_test_path: components.OutputPath("np_array"),  # noqa: F821
+    mlpipeline_ui_metadata_path: components.OutputPath(),
+    seed: int = 1337,
+):
+    import numpy as np
+    from sklearn import datasets
+    from sklearn.model_selection import train_test_split
+    import wandb
+    import json
+
+    def add_wandb_visualization(run, mlpipeline_ui_metadata_path):
+        """NOTE: To use this, you must modify your component to have an output called `mlpipeline_ui_metadata_path` AND call `wandb.init` yourself inside that component.
+
+        Example usage:
+
+        def my_component(..., mlpipeline_ui_metadata_path: OutputPath()):
+            import wandb
+            from wandb.integration.kfp.helpers import add_wandb_visualization
+
+            with wandb.init() as run:
+                add_wandb_visualization(run, mlpipeline_ui_metadata_path)
+
+                ... # the rest of your code here
+        """
+
+        def get_iframe_html(run):
+            return f'<iframe src="{run.url}?kfp=true" style="border:none;width:100%;height:100%;min-width:900px;min-height:600px;"></iframe>'
+
+        iframe_html = get_iframe_html(run)
+        metadata = {
+            "outputs": [
+                {"type": "markdown", "storage": "inline", "source": iframe_html}
+            ]
+        }
+
+        with open(mlpipeline_ui_metadata_path, "w") as metadata_file:
+            json.dump(metadata, metadata_file)
```

Comment:
We should be importing this from `wandb.integration.kfp.helpers`, but are not yet because of how tests get installed in kfp components.

Reply:
Is this ever going to change? :)

Reply:
I think it can change in the next version of SDK :)